lib/frame.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+class Frame
+  TEN_PINS = 10
+  FIRST_ROLL = 0
+  SECOND_ROLL = 1
+
+  def initialize
+    @rolls = []
+    @frame_type = nil
+  end
+
+  def log_roll(pins_downed)
+    raise 'Pins downed must be between 0 and 10' if invalid_pins?(pins_downed)
+    raise 'Frame complete. Cannot roll again' if frame_complete?
+
+    process_frame_type(pins_downed)
+    process_roll(pins_downed)
+  end
+
+  def sum_frame
+    @rolls.sum
+  end
+
+  def all_rolls
+    @rolls
+  end
```

Comment:
This method seems unnecessary to me, as the rolls can be accessed directly through the `@rolls` instance method.